src/main/java/com/squareup/javapoet/TypeVariableName.java

```diff
@@ -57,6 +57,18 @@ private TypeVariableName(String name, List<TypeName> bounds, List<AnnotationSpec
     return new TypeVariableName(name, bounds);
   }
 
+  public TypeVariableName withBounds(Type... bounds) {
+    return get(name, bounds);
```

Comment:
True. We always do order-preserving sets, but I was thinking of a case where you're potentially building up interface-based bounds. Might as well leave it as a list until someone complains with an actual use case.